pandas/io/pytables.py

```diff
@@ -455,7 +455,7 @@ def read_hdf(
             chunksize=chunksize,
             auto_close=auto_close,
         )
-    except (ValueError, TypeError, KeyError):
+    except (ValueError, TypeError, KeyError, IndexError):
```

Comment:
probably this should just be 
```suggestion
    except BaseException:
```

as when using `with HDFStore(...):` it will still close on all exceptions

Reply:
Yeah that makes sense (I'm not sure if we have a linting rule that may complain about `BaseException` but worth a try